File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.java
class Solution {
    public int solution(String[] babbling) {
        int answer = 0;

        for (String word : babbling) {
            word = word
                .replace("aya", "1")
                .replace("ye", "2")
                .replace("woo", "3")
                .replace("ma", "4");
            
            if (word.contains("11") ||
                word.contains("22") ||
                word.contains("33") ||
                word.contains("44")) {
                continue;
            }
            
            if (word.matches("[1234]+")) {
                answer++;
            }
        }

        return answer;
    }
}

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.kt
class Solution {
    fun solution(babbling: Array<String>): Int {
        var answer = 0
        
        for (word in babbling) {
            var replaced = word
                .replace("aya", "1")
                .replace("ye", "2")
                .replace("woo", "3")
                .replace("ma", "4")
                
            if (replaced.contains("11") ||
                replaced.contains("22") ||
                replaced.contains("33") ||
                replaced.contains("44")) {
                continue
            }

            if (replaced.matches(Regex("[1234]+"))) answer++
        }
        
        return answer
    }
}